todoList의 레이아웃 구현 (스타일 제외)

JungDaewook opened the pull request:
[구현 기능]
- todoList의 레이아웃 구현

[어려웠던 점]
- 코드를 짜다 보니 html, js, css 코드의 가독성이 떨어짐

[중점 리뷰 사항]
- html, css, js 코드의 가독성
- html 코드 불필요한 컨테이너와 태그

Commit:
변수명이 모호해서 변수명 변경.
todoItemIdCount -> todoItemIdNumber로 변경.

Commit:
디버깅 과정에서 생겨난 문자열 실수 수정

Commit:
코멘트에 따라 변수명 수정
1. todoItemIdNumber -> todoItemIdIndex

JungDaewook committed:
Merge pull request #13 from JungDaewook/refactoring2

Refactoring2

Commit:
body를 모두 포함하는 container 생성

Commit:
입력창과 추가, 전체 삭제 버튼 레이아웃 설정

Commit:
todoList 안의 li 요소들의 레이아웃 설정
1. js 파일에서 innerHTML 부분을 수정하여 버튼들을 감싸는 <div> 태그 추가
2. css 파일에서 todoList 안의 li 레이아웃을 설정하는 코드 추가

Commit:
레이아웃 간의 전체적인 흐름 구현
1. todoListContentContainer가 창을 축소해도 줄지 않는 현상이 발생하여 nav와 ad를 추가.
2. 전체적인 흐름을 보기 위해 코드 추가
3. 추후 색깔 변경 예정

JungDaewook commented on functionsOfTodoList.js:
```diff
@@ -14,8 +14,7 @@ function addTodoItem()  {
   todoItem.appendChild(textNode);
-  todoItem.innerHTML += "&nbsp; <input type = 'button' id = 'strikeButton' onclick = 'strikeTodoItem(\""+todoItemId+"\")' value = '완료'/>"; 
-  todoItem.innerHTML += "<input type = 'button' onclick = 'removeTodoItem(\""+todoItemId+"\")' value = '삭제'/>";
+  todoItem.innerHTML += "<div class = 'strikeAndRemoveButton'><input type = 'button' id = 'strikeButton' onclick = 'strikeTodoItem(\""+todoItemId+"\")' value = '완료'/><input type = 'button' onclick = 'removeTodoItem(\""+todoItemId+"\")' value = '삭제'/></div>"; 
```
[질문]
`todoItem.innerHTML += "<div class = 'strikeAndRemoveButton'>&nbsp; <input (생략) value = '완료'>";` 
`todoItem.innerHTML += "<input (생략) value = '삭제'/></div>";`
- 문제 상황 
위 코드와 같이 먼저 구현했는데 의도와는 다르게
`</div>` 태그가 
`<div class = 'strikeAndRemoveButton'>` 바로 뒤에  붙는 현상 발생.

- 예상 문제 원인 
브라우저에서 렌더링하는 과정에서 자동 교정을 통해 그런 현상이 발생하는 것 같음.

- 해결방안
한 행에 아주 긴 코드로 구현해 자동 교정 기능이 발동되지 않도록 구현

- 개선하고 싶은 점
한 행에 너무 긴 코드가 있다 보니 가독성이 떨어짐. 코드 가독성을 해치지 않고 자동 교정 기능이 발동되지 않도록 구현하는 방법 탐색 필요.

unpo88 replied:
@JungDaewook 

개선하고 싶은 점 잘 읽어보았고 🤔 

각 태그의 구성 요소를 변수화해서 선언할 수 있는 방법이 없을까요?

JungDaewook replied:
[답변]
해당 문제점은 #16 PR이 merge 된 이후에 해결해야 할 것 같아요.
 #16 에서 컨테이너를 삭제하는 등 코드에 많은 변화가 있어서 문제를 지금 해결한다면 두 PR 사이에 혼선을 불러 일으킬 것 같아서요.

unpo88 replied:
현재 PR은 그럼 Merge 되면 확인해보겠습니다~

JungDaewook replied:
[문제 해결]
https://github.com/JungDaewook/Todo-JavaScript/pull/16#issuecomment-1200105993
위 PR에서 해당 문제를 해결했어요

[이유]
#16 PR에서 merge가 지체되어 현재 PR이 아닌 #16 PR에서 문제를 해결했어요.

JungDaewook commented on styleOfTodoList.css:
```diff
@@ -1,19 +1,34 @@
+    height : 100vh;
     display : flex;
     flex-direction : column;
-    align-items : center;
```
[질문]
-문제 상황
`align-items : center;` 코드를 삭제하지 않으면 창을 축소했을 때 "todoListContentContainer"가 축소되지 않는 문제 발생

-해결 방법
`align-items : center;` 코드 삭제

- 개선점
`align-items : center;` 코드가 있을 때 왜 축소가 안 되고 삭제하면 왜 축소가 되는 지 이해 필요

JungDaewook replied:
https://user-images.githubusercontent.com/40848888/179003435-72808b12-65f2-47b7-82af-0a8ae627bedc.mov

JungDaewook replied:
https://user-images.githubusercontent.com/40848888/179003656-d1e5dae1-232b-4a88-ab6e-78a8bff434d1.mov

unpo88 replied:
@JungDaewook 

요 문제는 파악이 되었을까요?

JungDaewook replied:
[답변]
창이 줄어들어도 줄어들지 않는 문제가 발생했던 todoListContentContainer 가 bodyContainer 전에 todoListMainContainer로 한번 더 쌓여 있어서 발생한 문제였어요.

Commit:
코드 가독성을 위해 레이아웃과 스타일 부분의 코드 분리

Commit:
최상단 제목의 스타일 구현.

Commit:
todoListContentHeader 디자인 구현 완료
1. js 파일에서 추가버튼과 전체삭제 버튼에 id 부여
2. todoItem 입력창과 추가, 전체삭제 버튼 디자인 구현.

unpo88 commented on styleOfTodoList.css:
```diff
@@ -0,0 +1,38 @@
+.bodyContainer {
+    background-color : blue;
+    height : 100vh;
```
@JungDaewook 

`vh` 의 의미를 혹시 알고 사용했을까요!?

JungDaewook replied:
[답변]
[https://www.w3schools.com/cssref/css_units.asp](url)
해당 링크를 확인하여 뷰포트의 1%를 나타내는 단위라는 것을 알아보았어요. 

아래의 PR에서 bodyContainer에 vh 단위를 사용하니 생기는 문제에 대해 정리했고 해결했어요.
https://github.com/JungDaewook/Todo-JavaScript/pull/16#pullrequestreview-1040468506

unpo88 reviewed (approved):
@JungDaewook 

확인해주세요!

Commit:
#todoList 안의 요소들 디자인 구현 완료.
1. js의 innerHTML 코드에서 삭제 버튼의 id 부여
2. todoList 안의 요소들 디자인 구현.

Commit:
bodyContainer의 설정 일부를 body로 이동
1. bodyContainer와 body 사이에 하얀색으로 margin이 잡히는 문제 해결
2. todoItem을 많이 추가했을 때 아래 부분에 bodyContainer의 설정이 해결되지 않는 문제 해결

Commit:
추가, 완료 버튼에도 삭제, 전체삭제 버튼과 같이 hover 추가

Commit:
불필요한 코드 제거 및 css 파일 줄 나눔

unpo88 commented:
@JungDaewook 

요 PR은 어떤 상황일까요~?

Commit:
불필요한 컨테이너 제거 및 css 코드 수정

Commit:
css 코드 띄어쓰기 통일

JungDaewook commented:
> @JungDaewook
> 
> 요 PR은 어떤 상황일까요~?

https://github.com/JungDaewook/Todo-JavaScript/pull/16
[답변]
당초 스타일을 제외한 레이아웃을 구현하고, 수정하기 위한 PR이었는데, 리팩토링을 진행할 때는 스타일과 레이아웃을 따로 구현하는 것이 비효율적이라고 판단하여 위 PR에서 레이아웃에 관련된 코드까지 한번에 수정하고 있어요. 수정 작업이 끝나면 이 브랜치로 merge 한 후 이 브랜치를 main 브랜치로 merge할 예정이에요.

unpo88 commented:
@JungDaewook 

> [답변]
> 당초 스타일을 제외한 레이아웃을 구현하고, 수정하기 위한 PR이었는데, 리팩토링을 진행할 때는 스타일과 레이아웃을 따로 구현하는 것이 비효율적이라고 판단하여 위 PR에서 레이아웃에 관련된 코드까지 한번에 수정하고 있어요. 수정 작업이 끝나면 이 브랜치로 merge 한 후 이 브랜치를 main 브랜치로 merge할 예정이에요.

요약하면 [Style without layout of todo list #16](https://github.com/JungDaewook/Todo-JavaScript/pull/16) PR이 모두 완료되어야 순차적으로 Merge를 한다는 이야기죠?

[피드백]
두괄식 표현으로 이야기해보는 습관을 가지면 좋을 것 같습니다.
설명이 매우 길어서 어떤 맥락을 이야기하는 것인지 이해하기 어려움이 있어요 😢 

----------------

상단에 제가 남겨놓은 코멘트들에 대해서는 답변이 없어요 🤔 

답변이 필요하지 않을까요? 코멘트 확인을 한 것인지 어떤 상황인지 제가 인지할 방법이 없어요.

Commit:
불필요한 개행 제거

Commit:
li 요소에 marker를 추가하기 위한 css 코드 수정.

Commit:
marker 위치 조정을 위한 코드 삭제 및 추가
1. list-style: none; 코드로 marker 삭제
2. ::before 를 이용해 위치를 조정할 수 있는 marker(속성은 before인) 생성.

Commit:
innerHTML 코드 가독성 문제 해결
1. 완료, 취소, 삭제 버튼의 innerHTML 내용을 return하는 함수 생성

JungDaewook commented:
> @JungDaewook
> 
> > [답변]
> > 당초 스타일을 제외한 레이아웃을 구현하고, 수정하기 위한 PR이었는데, 리팩토링을 진행할 때는 스타일과 레이아웃을 따로 구현하는 것이 비효율적이라고 판단하여 위 PR에서 레이아웃에 관련된 코드까지 한번에 수정하고 있어요. 수정 작업이 끝나면 이 브랜치로 merge 한 후 이 브랜치를 main 브랜치로 merge할 예정이에요.
> 
> 요약하면 [Style without layout of todo list #16](https://github.com/JungDaewook/Todo-JavaScript/pull/16) PR이 모두 완료되어야 순차적으로 Merge를 한다는 이야기죠?
> 
> [피드백] 두괄식 표현으로 이야기해보는 습관을 가지면 좋을 것 같습니다. 설명이 매우 길어서 어떤 맥락을 이야기하는 것인지 이해하기 어려움이 있어요 😢

[답변 수정]
- 코멘트에 따라 두괄식으로 답변을 수정해봤어요

[답변]
- #16 PR에서 현재 PR로 merge한 후 나머지 코멘트에 의해 수정할 예정이에요

[이유]
- #16 PR에서 스타일과 레이아웃 모두를 수정해야 하는 상황이 생겨서 현재 상황에서 이 PR을 수정한다면 conflict가 날 것으로 예상이 되기 때문이에요.

Commit:
코멘트에 따라 불필요한 코드 수정 및 축약
1. 불필요한 코드 제거
2. 축약 가능한 코드 축약
3. li 안의 요소들 수직 방향으로 가운데 정렬

Commit:
불필요한 코드 제거

Commit:
fonctionsOfTodoList.js 파일에서 버튼 관련 코드 중복 문제 해결
1. 버튼을 만들기 위한 innerHtml text를 div 태그로 감싸는 함수 구현
2. 함수 내에서 버튼을 만들기 위한 innerHtml text를 변수로 설정.

Commit:
wrapTextForInnerHtmlWithDivTag함수의 함수명을 함수 역할을 더 명확하게 드러내는 함수명으로 변경

Commit:
구현과정에서 실수로 남아있던 코드 제거

JungDaewook committed:
Merge pull request #16 from JungDaewook/styleWithoutLayoutOfTodoList

Style without layout of todo list
